pkg/scheduler/algorithm/priorities/selector_spreading.go

```diff
@@ -84,26 +84,9 @@ func (s *SelectorSpread) CalculateSpreadPriorityMap(pod *v1.Pod, meta interface{
 		}, nil
 	}
 
-	count := int(0)
-	for _, nodePod := range nodeInfo.Pods() {
-		if pod.Namespace != nodePod.Namespace {
-			continue
-		}
-		// When we are replacing a failed pod, we often see the previous
-		// deleted version while scheduling the replacement.
-		// Ignore the previous deleted version for spreading purposes
-		// (it can still be considered for resource restrictions etc.)
-		if nodePod.DeletionTimestamp != nil {
-			klog.V(4).Infof("skipping pending-deleted pod: %s/%s", nodePod.Namespace, nodePod.Name)
-			continue
-		}
-		for _, selector := range selectors {
-			if selector.Matches(labels.Set(nodePod.ObjectMeta.Labels)) {
-				count++
-				break
-			}
-		}
-	}
+	filteredPods := filterPods(pod.Namespace, selectors, nodeInfo)
```

Comment:
This unnecessarily allocates a slice of pods. We only need the number of matching pods here and are not interested in the pods themselves. Memory allocation/release is known to cause performance degradation and should be avoided when possible.

Reply:
Thanks. Fixed.